crates/router/src/core/payments/helpers.rs

```diff
@@ -1360,6 +1342,36 @@ pub async fn make_pm_data<'a, F: Clone, R>(
     Ok((operation, payment_method))
 }
 
+pub async fn store_in_vault_and_generate_ppmt(
+    state: &AppState,
+    payment_method_data: &api_models::payments::PaymentMethodData,
+    payment_intent: &payment_intent::PaymentIntent,
+    payment_attempt: &payment_attempt::PaymentAttempt,
+    payment_method: enums::PaymentMethod,
+) -> RouterResult<String> {
+    let router_token = vault::Vault::store_payment_method_data_in_locker(
+        state,
+        None,
+        payment_method_data,
+        payment_intent.customer_id.to_owned(),
+        payment_method,
+    )
+    .await?;
+    let parent_payment_method_token = generate_id(consts::ID_LENGTH, "token");
+    let key_for_hyperswitch_token = payment_attempt.payment_method.map(|payment_method| {
```

Comment:
Please check why `key_for_hyperswitch_token` is optional?
We are about to store in vault, so we should always get the key.

So, we have access to pmd, try getting pm from it and then generate the key